common/elasticsearch/client/v6/client_test.go

```diff
@@ -0,0 +1,149 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package v6
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/olivere/elastic"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/uber/cadence/common/config"
+	"github.com/uber/cadence/common/log/testlogger"
+)
+
+func TestNewV6Client(t *testing.T) {
+	logger := testlogger.New(t)
+	testServer := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer testServer.Close()
+	url, _ := url.Parse(testServer.URL)
+	connectionConfig := &config.ElasticSearchConfig{
+		URL:                *url,
+		DisableSniff:       true,
+		DisableHealthCheck: true,
+	}
+	sharedClient := testServer.Client()
+	client, err := NewV6Client(connectionConfig, logger, sharedClient, sharedClient)
+	assert.NoError(t, err)
+	assert.NotNil(t, client)
+
+	// failed case due to an unreachable Elasticsearch server
+	badURL, _ := url.Parse("http://nonexistent.elasticsearch.server:9200")
+	connectionConfig.DisableHealthCheck = false
+	connectionConfig.URL = *badURL
+	_, err = NewV6Client(connectionConfig, logger, nil, nil)
+	assert.Error(t, err)
+}
+
+func TestCreateIndex(t *testing.T) {
+	testServer := getTestServer()
+	defer testServer.Close()
+	// Create a new MockESClient
+	mockClient, err := elastic.NewClient(
+		elastic.SetURL(testServer.URL),
+		elastic.SetSniff(false),
+		elastic.SetHealthcheck(false),
+		elastic.SetHttpClient(testServer.Client()),
+	)
+	assert.NoError(t, err)
+
+	elasticV6 := ElasticV6{
+		client: mockClient,
+	}
+	err = elasticV6.CreateIndex(context.Background(), "testIndex")
+	assert.NoError(t, err)
+}
+
+func TestPutMapping(t *testing.T) {
+	testServer := getTestServer()
+	defer testServer.Close()
+	// Create a new MockESClient
+	mockClient, err := elastic.NewClient(
+		elastic.SetURL(testServer.URL),
+		elastic.SetSniff(false),
+		elastic.SetHealthcheck(false),
+		elastic.SetHttpClient(testServer.Client()))
+	assert.NoError(t, err)
+
+	elasticV6 := ElasticV6{
+		client: mockClient,
+	}
+	err = elasticV6.PutMapping(context.Background(), "testIndex", `{
```

Comment:
you can validate whether the test server received a specific http request (path and payload validation)